### PR TITLE
:seedling: update talos/sidero references

### PR DESCRIPTION
**Pull request by rsmitty**
**What this PR does / why we need it**:

Talos Systems is now Sidero Labs and we just renamed our github org.
This PR updates all of the references to make sure they align with the
new org name. Saw an issue where clusterctl couldn't follow the
redirects to the yaml assets, which appears to also occur when manually
curling (assuming a github bug).

Signed-off-by: Spencer Smith <spencer.smith@talos-systems.com>

### CODE REVIEW

**Review by smira — APPROVED** [review]

**Review by frezbo — APPROVED** [review]

**Comment by sbueringer** [comment]
@rsmitty Thx for the PR. 

Lgtm from my side when the unit test is fixed (looks like some issue with the diff)

**Comment by killianmuldoon** [comment]
/lgtm

(pending unit test)

**Comment by sbueringer** [comment]
Perfect, thx! :)

/lgtm

The change looks to me like we should backport to v1.1. @rsmitty WDYT?

**Comment by rsmitty** [comment]
yep, will do after this is merged.

**Comment by sbueringer** [comment]
/cherry-pick release-1.1

@rsmitty you don't have to do it manually :)

**Pull request merged**
